### ISSUE TITLE
[ char in tablename makes Refine unable to save in Excel format

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Illablu/google-refine#574]
```
I guess I've found a bug in Google Refine (last stable version for mac).

If you want to export a table in excel format and its name contains a special 
char such as "[", google refine crashes and doesn't save it.

```

Original issue reported on code.google.com by `distant....@gmail.com` on 7 May 2012 at 5:59

**Comment by GoogleCodeExporter** [comment]
```
Could you please name the Google Refine version + iOS version + an example of a 
name that crashes?

```

Original comment by `j...@tekii.com.ar` on 8 May 2012 at 1:26

**Comment by GoogleCodeExporter** [comment]
```
It's confirmed also for refine 2.5 in Linux.
It seems that spetial characters are not replaced before saving the file in 
XlsExporter.java: 83 
And poi, does not support this in the filename:

    never null
    minimum length is 1
    maximum length is 31
    doesn't contain special chars: : 0x0000, 0x0003, / \ ? * ] [
    Sheet names must not begin or end with ' (apostrophe)

http://poi.apache.org/apidocs/org/apache/poi/ss/util/WorkbookUtil.html

I upload a posible patch for this.
```

Original comment by `j...@tekii.com.ar` on 8 May 2012 at 2:17

Attachments:
- [0001-Using-WorkbookUtil-for-creating-a-safe-name.patch](https://storage.googleapis.com/google-code-attachments/google-refine/issue-574/comment-2/0001-Using-WorkbookUtil-for-creating-a-safe-name.patch)

**Comment by GoogleCodeExporter** [comment]
Original comment by `tfmorris` on 18 Sep 2012 at 8:11
- Changed state: **Accepted**
- Added labels: **Milestone-2.6**

**Comment by GoogleCodeExporter** [comment]
Original comment by `tfmorris` on 12 Oct 2012 at 7:16
- Added labels: **Patch-NeedsReview**

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the patch!  Patch applied in r2582.  Look for the fix in the 2.6 
release.
```

Original comment by `tfmorris` on 12 Oct 2012 at 11:06
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `tfmorris` on 12 Oct 2012 at 11:07
- Added labels: **Patch-Applied**
- Removed labels: **Patch-NeedsReview**
